arch/arc/core/reset.S

```diff
@@ -16,6 +16,9 @@
 #include <zephyr/arch/cpu.h>
 #include <swap_macros.h>
 #include <zephyr/arch/arc/asm-compat/assembler.h>
+#ifdef CONFIG_SOC_ARC_HSDK4XD
```

Comment:
I'd make this more generic. I.e. by not using exact SoC Kconfig and by replacing it with "CONFIG_EARLY_SOC_INIT" or something like that.

Reply:
In generic way this will require to create "soc_ctrl.h" header for each SOC. Is it fine?

Reply:
We still can guard this include under ifdef. But we can do ifdef against generic Kconfig symbol instead of SoC specific one.

Reply:
please check the recent changes